hedera-node/hedera-app/src/test/java/com/hedera/node/app/signature/AppSignatureVerifierTest.java

```diff
@@ -0,0 +1,177 @@
+import com.hedera.hapi.node.base.ThresholdKey;
+import com.hedera.node.app.spi.signatures.SignatureVerification;
+import com.hedera.node.app.spi.signatures.SimpleKeyCount;
+import com.hedera.node.app.spi.signatures.SimpleKeyVerification;
+import com.hedera.node.config.data.HederaConfig;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class AppSignatureVerifierTest {
+    private static final Key A_KEY = Key.newBuilder()
+            .ed25519(Bytes.fromHex("0101010101010101010101010101010101010101010101010101010101010101"))
+            .build();
+    private static final Key B_KEY = Key.newBuilder()
+            .ecdsaSecp256k1(Bytes.fromHex("020202020202020202020202020202020202020202020202020202020202020202"))
+            .build();
+    private static final Key C_KEY = Key.newBuilder()
+            .contractID(ContractID.newBuilder().contractNum(666L).build())
+            .build();
+    private static final Key TEST_KEY = Key.newBuilder()
+            .thresholdKey(ThresholdKey.newBuilder().threshold(1).keys(new KeyList(List.of(A_KEY, B_KEY, C_KEY))))
+            .build();
+
+    @Mock
+    private SignatureExpander signatureExpander;
+
+    @Mock
+    private com.hedera.node.app.signature.SignatureVerifier signatureVerifier;
+
+    @Mock
+    private SignatureVerification verification;
+
+    @Mock
+    private SignatureVerificationFuture future;
+
+    private AppSignatureVerifier subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new AppSignatureVerifier(
+                DEFAULT_CONFIG.getConfigData(HederaConfig.class), signatureExpander, signatureVerifier);
+    }
+
+    @Test
+    void usesDefaultKeyVerifierWithNoOverride() throws ExecutionException, InterruptedException, TimeoutException {
+        final Set<ExpandedSignaturePair> expandedPairs =
+                Set.of(new ExpandedSignaturePair(A_KEY, A_KEY.ed25519OrThrow(), null, SignaturePair.DEFAULT));
+        willAnswer(invocationOnMock -> {
+                    final Set<ExpandedSignaturePair> pairs = invocationOnMock.getArgument(2);
+                    pairs.addAll(expandedPairs);
+                    return null;
+                })
+                .given(signatureExpander)
+                .expand(eq(TEST_KEY), eq(SignatureMap.DEFAULT.sigPair()), any(Set.class));
+
+        final var hederaConfig = DEFAULT_CONFIG.getConfigData(HederaConfig.class);
+        given(future.get(hederaConfig.workflowVerificationTimeoutMS(), TimeUnit.MILLISECONDS))
+                .willReturn(verification);
```

Comment:
Seems appropriate for the `@BeforeEach`?

Possibly also L87-95 - in this set of tests the same expanded map is used in each test but in the future maybe not?

Reply:
Good point, created a `@Nested` class where the universal setup is in a `@BeforeEach`.